client/post-editor/editor-featured-image/index.jsx

```diff
@@ -115,6 +127,7 @@ export default React.createClass( {
 				borderless
 				compact
 				className={ classes }>
+				{ site && featuredImageId && <QueryMedia siteId={ site.ID } mediaId={ featuredImageId } /> }
```

Comment:
Although `mediaId` is not a required prop in `QueryMedia`, it's necessary to check for `featuredImageId` here otherwise you'll receive a console error when `featuredImageId` is `null` (i.e. when you remove a featured image from a post).